plugins/development/plugin-git/src/node/gitPlugin.ts

```diff
@@ -60,25 +41,63 @@ export const gitPlugin =
           return
         }
 
+        if (filter && !filter(page)) return
+
+        const { frontmatter } = page
+
+        if (
+          !(frontmatter.changelogs ?? contributors) &&
+          !(frontmatter.contributors ?? changelogs) &&
+          !createdTime &&
+          !updatedTime
+        ) {
+          return
+        }
+
         const filePaths = [
           page.filePathRelative,
           ...(page.frontmatter.gitInclude ?? []).map((item) =>
             path.join(page.filePathRelative, '..', item),
           ),
         ]
 
+        const commits = await getCommits(filePaths, cwd)
+
+        if (commits.length === 0) return
+
         if (createdTime !== false) {
-          page.data.git.createdTime = await getCreatedTime(filePaths, cwd)
+          page.data.git.createdTime = commits[commits.length - 1].date
         }
 
         if (updatedTime !== false) {
-          page.data.git.updatedTime = await getUpdatedTime(filePaths, cwd)
+          page.data.git.updatedTime = commits[0].date
         }
 
-        if (contributors !== false) {
-          const result = await getContributors(filePaths, cwd)
+        if ((frontmatter.contributors ?? contributors) !== false) {
+          const options = isPlainObject(contributors) ? contributors : {}
+          options.transform ??= transformContributors
+          page.data.git.contributors = await resolveContributors(
+            commits,
+            options,
+            gitType,
+            Array.isArray(frontmatter.contributors)
+              ? frontmatter.contributors
+              : [],
+          )
+        }
 
-          page.data.git.contributors = transformContributors?.(result) ?? result
+        if ((frontmatter.changelogs ?? changelogs) !== false) {
```

Comment:
```suggestion
        if (frontmatter.changelogs ?? changelogs ?? true) {
```

I'm reviewing the code on mobile phone, I think it's equivalent and you'd better double check it